smarts/core/mission_planner.py

```diff
@@ -193,6 +200,58 @@ def _edge_ids(self, pose: Pose, lane_id: str = None):
 
         return edge_ids
 
+    def cut_in_waypoints(self, pose: Pose, sim, vehicle):
+        radius = 30
+        neighborhood_vehicles = sim.neighborhood_vehicles_around_vehicle(
+            vehicle=vehicle, radius=radius
+        )
+
+        if not neighborhood_vehicles:
+            return []
+
+        nei_vehicle = neighborhood_vehicles[0]
+        position = pose.position[:2]
+        lane = self._road_network.nearest_lane(position)
+        target_position = nei_vehicle.pose.position[:2]
+        target_lane = self._road_network.nearest_lane(target_position)
+
+        offset = self._road_network.offset_into_lane(lane, position)
+        target_offset = self._road_network.offset_into_lane(
+            target_lane, target_position
+        )
+        if offset < target_offset:
+            # Need to catch up with the target vehicle first
+            return []
+
+        nei_wps = self._waypoints.waypoint_paths_on_lane_at(
+            target_position, target_lane.getID(), 60
+        )
+        p0 = pose.position[:2]
+        p1 = nei_wps[0][len(nei_wps[0]) // 2].pos
+        p2 = target_position[:2]
+        p3 = nei_wps[0][-1].pos
+        p_x, p_y = bezier([p0, p1, p2, p3], 20)
+        trajectory = []
+        prev = position[:2]
+        for i in range(len(p_x)):
+            pos = np.array([p_x[i], p_y[i]])
```

Comment:
No need to assign these, just put em directly in the Waypoint initializer since it's using named args already.

Reply:
Oh, this is because I need to calculate the heading between prev position and current position.